Source/OCMockTests/OCMockObjectMacroTests.m

```diff
@@ -477,4 +477,27 @@ - (void)testCanUseMacroToStubMethodWithAnyNonObjectArgumentChainedWithOCMStubRec
     XCTAssertEqualObjects(@"f", [mock commonPrefixWithString:@"foo" options:NSCaseInsensitiveSearch]);
 }
 
+- (void)testInitWithRecorderMethodsBeingPassedThroughMock
+{
+    // Because of the way the macros work, you can call recorder methods on the mock and they
+    // will work correctly. Technically these are a mix of old syntax and new.
+    // If these test fail we expect a crash.
+    // Note that the andReturn:nil has to be the first stub so that there is a return value.
```

Comment:
I don't understand this. Sorry, I tried.

Reply:
If you don't have the `andReturn:nil` as the first stub you will get 
```
error: -[OCMockObjectMacroTests testInitWithRecorderMethodsBeingPassedThroughMock] : failed: caught "NSInvalidArgumentException", "initWithString: was stubbed but no return value set. A return value is required for an init method. If you intended to return nil, make this explicit with .andReturn(nil)"
```

An alternative is to add `andReturn:nil` to each of the tests (or `.andReturn(nil)`) but I thought it made the tests even more confusing than they already are.

Reply:
Thanks. Now I get it.